include/morphio/endoplasmic_reticulum.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <morphio/types.h>
+
+namespace morphio {
+/**
+ * The entry-point class to access endoplasmic reticulum data
+ *
+ * Spec https://bbpteam.epfl.ch/documentation/projects/Morphology%20Documentation/latest/h5v1.html
+ **/
+class EndoplasmicReticulum
+{
+public:
+    /**
+       Returns the list of neuronal section indices
+    **/
+    const std::vector<uint32_t>& sectionIndices() const;
+
+    /**
+       Returns the volumes for each neuronal section
+    **/
+    const std::vector<float>& volumes() const;
+
+    /**
+       Returns the surface areas for each neuronal section
+    **/
+    const std::vector<float>& surfaceAreas() const;
+
+    /**
+       Returns the number of filaments for each neuronal section
+    **/
+    const std::vector<uint32_t>& filamentCounts() const;
+
+private:
+    EndoplasmicReticulum(std::shared_ptr<Property::Properties> properties)
```

Comment:
I'm still don't fully get why this is private, and then you have to declare `Morphology` a friend.  Seems like you're allowing `Morphology` much more access (ie: to _properties) than is necessary.

Reply:
> I'm still don't fully get why this is private, and then you have to declare Morphology a friend. 

My idea was that in MorphIO the *only* public constructor would be `Morphology`.
I don't want people to start creating EndoplasmicReticulum, or Section, Mitochondria objects that are not attached to a `Morphology` object.
You should only access those objects through `Morphology`:
```
m = Morphology("my-morph.swc");
m.mitochondria().do_stuff();
m.section(1).do_other_stuff();
m.endoplasmicReticulum().do_even_more_stuff();
```

> Seems like you're allowing Morphology much more access (ie: to _properties) than is necessary.

She only place where the EndoplasmicReticulum is called is in `Morphology::endoplasmicReticulum()` and the passed `_properties` structure is already owned by `Morphology`

Does this make sense ?

Reply:
Ok, I think I get it